services/node_manager/src/main/java/com/futurewei/alcor/nodemanager/controller/NodeController.java

```diff
@@ -49,7 +54,22 @@
 
     @RequestMapping(
             method = POST,
-            value = {"/nodes/upload", "/v4/nodes/upload"})
+            value = {"/bulk/nodes", "/v4/bulk/nodes"})
+    public List<NodeInfo> registerNodesInfoBulk(@RequestBody BulkNodeInfoJson resource) throws Exception {
+        if (resource == null) {
+            throw new ParameterNullOrEmptyException(NodeManagerConstant.NODE_EXCEPTION_FILE_EMPTY);
+        }
+        try {
+            return service.createNodeInfoBulk(resource.getNodeInfos());
+        } catch (Exception e) {
+            LOG.log(Level.SEVERE,e.getMessage());
+            throw e;
+        }
+    }
+
+    @RequestMapping(
+            method = POST,
+            value = {"/bulk/nodes", "/v4/bulk/nodes"})
```

Comment:
Could you please check API name?  "/bulk/nodes"? or "/nodes/bulk"
API naming convention was: 
 /module_name (e.g. "nodes" for node manage")/api_name ("bulk")